buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/GenerateCoveragePercentsTask.groovy

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle.plugins.measurement
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+
+public class GenerateCoveragePercentsTask extends DefaultTask {
+
+    @OutputFile
+    File reportFile
+
+    @Input
+    List<String> packages
+
+    XmlSlurper parser
+
+    GenerateCoveragePercentsTask() {
+        parser = new XmlSlurper()
+        parser.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false)
+        parser.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false)
+    }
+
+    @TaskAction
+    def generate() {
+        // We would not have a pull request number unless running on Prow.
+        // Using 777 as a place holder when running locally for purpose such as debugging.
+        def pullRequestNumber = project.properties["pull_request"] ?: 777
+        generateJson(pullRequestNumber, packages)
+    }
+
+    private def getCoveragePercentFromReport(path) {
+        println "Parsing coverage file: $path"
+        def report = parser.parse(path)
+        def name = report.@name
+
+        def lineCoverage = report.counter.find {it.@type == 'LINE'}
+        def covered = Double.parseDouble(lineCoverage.@covered.text())
+        def missed = Double.parseDouble(lineCoverage.@missed.text())
+        def percent = covered / (covered + missed)
+
+        return new Tuple(name, percent)
+    }
+
+
+    private def generateJson(pullRequestNumber, packages) {
+        def coverages = [:]
+
+        // TODO(yifany@): Consolidate mappings with apksize and iOS.
+        def sdkMap = [
+                'firebase-common':1,
+                'firebase-database':2,
+                'firebase-database-collection':3,
+                'firebase-firestore':4,
+                'firebase-functions':5,
+                'firebase-inappmessaging-display':6,
+                'firebase-storage':7
+        ]
+
+        for (String p: packages) {
+            def path = p + '/build/reports/jacoco/checkCoverage/checkCoverage.xml'
```

Comment:
There's a better way to do this. You can use the Gradle Project objects and obtain the build directories from them. See my comment on the Gradle file.

Reply:
Done.